### PR TITLE
Product fee support

### LINKED ISSUE #866: Add support for product fees in iov-bns
One piece that was not completed in #856

This is needed for registering user names in the wallet, but not blocking updating the faucet

### PR DESCRIPTION
**Pull request by ethanfrey**
Closes #866 

getFeeQuote() will also query the product fees for the specified transaction kind and use that instead of the minimum anti-spam fee if present.

* Updated the bnsd CI genesis file to add a 5 CASH product fee for registering username
* Added logic to query these values and override other quote if needed
* Updated tests

### CODE REVIEW

File: packages/iov-bns/src/bnsconnection.ts [hunk 1]
```diff
@@ -613,15 +614,12 @@ export class BnsConnection implements BcpAtomicSwapConnection {
...
+    const product = await this.getProductFee(transaction.kind);
     // take maximum
-    const res = await this.query("/", Encoding.toAscii("gconf:cash:minimal_fee"));
-    if (res.results.length !== 1) {
-      throw new Error("Received unexpected number of fees");
-    }
-    const data = Encoding.fromAscii(res.results[0].value);
-    const amount = decodeJsonAmount(data);
-    return { tokens: amount };
+    const fee = product || min;
```
Comment (webmaster128) [inline packages/iov-bns/src/bnsconnection.ts]: I can see the elegance of this code, but in this case we should save the `getMinimumFee` call when `getProductFee` returns something.

File: packages/iov-bns/src/bnsconnection.spec.ts [hunk 2]
```diff
@@ -86,6 +86,12 @@ describe("BnsConnection", () => {
     fractionalDigits: 9,
     tokenTicker: cash,
   };
+  // this is enough money in an accout that registers names... twice the cost of one name registration product fee
```
Comment (webmaster128) [inline packages/iov-bns/src/bnsconnection.spec.ts]: accout -> account

File: packages/iov-bns/src/bnsconnection.ts (hunk 1 above)
Comment (webmaster128) [reply packages/iov-bns/src/bnsconnection.ts]: This implementation also means that the "min fee" is not a minimum fee but a default fee. In the case procuct fee < min fee, the fee is less than min fee.

**Comment by ethanfrey** [comment]
@webmaster128 I did the cleanup, should be good to merge

**Pull request merged**